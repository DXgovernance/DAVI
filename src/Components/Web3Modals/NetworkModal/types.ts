export interface NetworkModalProps {
  isOpen: boolean;
  onClose: () => void;
}
