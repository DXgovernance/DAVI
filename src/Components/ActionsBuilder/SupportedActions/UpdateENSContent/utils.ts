import contenthash from 'content-hash';
import { MAINNET_ID, LOCALHOST_ID, GOERLI_ID } from 'utils';
import { utils } from 'ethers';

const DEFAULT_NAMEHASH =
  '0x0000000000000000000000000000000000000000000000000000000000000000';

export const convertToNameHash = (name: string) => {
  if (!name) {
    return DEFAULT_NAMEHASH;
  }
  return utils.namehash(name);
};

export const convertToContentHash = (ipfsHash: string) => {
  if (!ipfsHash) {
    return '';
  }
  return contenthash.fromIpfs(ipfsHash);
};

export const convertToIpfsHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const getIpfsUrl = (ipfsHash: string) => {
  const ipfsRoot = 'ipfs://';
  return `${ipfsRoot}${ipfsHash}`;
};

export const isValidChainId = (chainId: number) => {
  if (!chainId || chainId === LOCALHOST_ID) {
    return MAINNET_ID;
  } else {
    return chainId;
  }
};

export const isAvailableOnENS = (chainId: number) => {
  const validChainId = isValidChainId(chainId);
  const ensNetworks = [MAINNET_ID, GOERLI_ID];
  return ensNetworks.includes(validChainId);
};
