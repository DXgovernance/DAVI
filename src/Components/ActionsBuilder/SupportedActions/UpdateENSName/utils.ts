import namehash from 'eth-ens-namehash';
import contenthash from 'content-hash';
import { useState, useEffect } from 'react';

export const convertToNameHash = (name: string) => namehash.hash(name);

export const convertToContentHash = (ipfsHash: string) =>
  contenthash.fromIpfs(ipfsHash);

export const convertToIPFSHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const useDebounce = (value: string, delay: number) => {
  const [debouncedValue, setDebouncedValue] = useState(value);

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);

    return () => {
      clearTimeout(handler);
    };
  }, [value, delay]);

  return debouncedValue;
};
