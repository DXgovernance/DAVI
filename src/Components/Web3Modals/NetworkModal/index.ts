export { default as NetworkModal } from './NetworkModal';
