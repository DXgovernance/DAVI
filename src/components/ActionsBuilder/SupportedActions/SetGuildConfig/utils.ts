import { FIELDS } from './constants';
import { BigNumber } from 'ethers';
import { SetGuildConfigFields, ControlField } from './types';

export const bn = (n: string | number | BigNumber) => BigNumber.from(n);

type UpdatedValuesReturn = {
  [key in ControlField]: {
    newValue: BigNumber;
    currentValue: BigNumber;
  };
};
export const getUpdatedValues = (
  current: SetGuildConfigFields,
  modifyed: SetGuildConfigFields
): UpdatedValuesReturn | {} => {
  if (!current || !modifyed) return {};
  return Object.keys(current).reduce((acc, key) => {
    if (!FIELDS.map(f => f.name).includes(key)) return acc;
    const currentValue = bn(current[key] || 0);
    const newValue = bn(modifyed[key] || 0);

    if (!currentValue.eq(newValue)) {
      return {
        ...acc,
        [key]: { newValue, currentValue },
      };
    }
    return acc;
  }, {});
};

export const pickUpdatedOrDefaultValue = (
  current: BigNumber,
  modifyed: BigNumber
): BigNumber => {
  if (BigNumber.isBigNumber(current)) {
    if (BigNumber.isBigNumber(modifyed) && !bn(modifyed).eq(current)) {
      return bn(modifyed);
    }
    return bn(current);
  } else if (BigNumber.isBigNumber(modifyed)) {
    return bn(modifyed);
  }
  return bn(0);
};
