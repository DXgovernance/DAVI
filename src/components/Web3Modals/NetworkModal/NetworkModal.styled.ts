import styled from 'styled-components';
export const Wrapper = styled.div`
  ${({ theme }) => theme.flexColumnNoWrap}
  margin: 0;
  padding: 0;
  width: 100%;
  background-color: ${({ theme }) => theme.bg1Color};
  border-radius: 10px;
`;

export const ContentWrapper = styled.div`
  background-color: ${({ theme }) => theme.colors.bg1};
  color: var(--body-text);
  padding: 2rem;
  ${({ theme }) => theme.mediaWidth.upToMedium`padding: 1rem`};
`;

export const UpperSection = styled.div`
  position: relative;
  background-color: var(--panel-background);

  h5 {
    margin: 0;
    margin-bottom: 0.5rem;
    font-size: 1rem;
    font-weight: 400;
  }

  h5:last-child {
    margin-bottom: 0px;
  }

  h4 {
    margin-top: 0;
    font-weight: 500;
  }
`;

export const OptionGrid = styled.div`
  display: flex;
  flex-direction: column;
`;
