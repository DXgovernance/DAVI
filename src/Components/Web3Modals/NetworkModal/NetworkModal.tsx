import { useWeb3React } from '@web3-react/core';

import { useRpcUrls } from 'provider/providerHooks';
import { getChains } from 'provider/connectors';
import { Modal } from 'old-components/Guilds/common/Modal';
import useNetworkSwitching from 'hooks/Guilds/web3/useNetworkSwitching';
import { getChainIcon } from 'utils';
import { Option } from '../components';
import {
  Wrapper,
  ContentWrapper,
  UpperSection,
  OptionGrid,
} from './NetworkModal.styled';
import { NetworkModalProps } from './types';

const NetworkModal: React.FC<NetworkModalProps> = ({ isOpen, onClose }) => {
  const { chainId } = useWeb3React();
  const rpcUrls = useRpcUrls();
  const { trySwitching } = useNetworkSwitching();

  return (
    <Modal
      header={<div>Switch network</div>}
      isOpen={isOpen}
      onDismiss={onClose}
      maxWidth={380}
    >
      <Wrapper>
        <UpperSection>
          <ContentWrapper>
            {rpcUrls && (
              <OptionGrid>
                {getChains(rpcUrls).map(chain => (
                  <Option
                    onClick={() => trySwitching(chain).then(onClose)}
                    key={chain.name}
                    icon={getChainIcon(chain.id)}
                    active={chain.id === chainId}
                    header={chain.displayName}
                  />
                ))}
              </OptionGrid>
            )}
          </ContentWrapper>
        </UpperSection>
      </Wrapper>
    </Modal>
  );
};

export default NetworkModal;
