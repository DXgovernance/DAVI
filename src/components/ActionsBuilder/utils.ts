import { DecodedAction } from './types';
import { Permission } from './types';
import Web3 from 'web3';
import { NULL_SIGNATURE } from 'utils';
import { SupportedAction } from './types';

const web3 = new Web3();

export const getPermissionArgs = (
  decodedActions: DecodedAction[]
): Permission[] => {
  const permissions = decodedActions.map(decodedAction => {
    // Native transfer of ETH does not have a function
    if (
      decodedAction?.decodedCall?.callType ===
        SupportedAction.NATIVE_TRANSFER ||
      decodedAction?.decodedCall?.callType === SupportedAction.RAW_TRANSACTION
    ) {
      return {
        from: decodedAction?.decodedCall?.from,
        to: decodedAction?.decodedCall?.to,
        callType: decodedAction?.decodedCall?.callType,
        functionSignature: NULL_SIGNATURE,
      };
    }
    const functionInputs = decodedAction?.decodedCall?.function?.inputs?.map(
      input => input.type
    );
    const functionName = decodedAction?.decodedCall?.function?.name;
    const functionSignature = web3.eth.abi.encodeFunctionSignature(
      `${functionName}(${functionInputs.join(',')})`
    );
    return {
      from: decodedAction?.decodedCall?.from,
      to: decodedAction?.decodedCall?.to,
      callType: decodedAction?.decodedCall?.callType,
      functionSignature,
    };
  });
  return permissions;
};
