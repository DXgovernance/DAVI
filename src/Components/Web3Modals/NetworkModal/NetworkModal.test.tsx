import NetworkModal from './NetworkModal';
import { render } from 'utils/tests';

jest.mock('contexts/index', () => jest.fn());

jest.mock('provider/connectors', () => {
  return {
    getChains: () => {
      return [
        {
          id: 1,
          name: 'mainnet',
          displayName: 'Ethereum Mainnet',
          defaultRpc: 'https://eth-mainnet.alchemyapi.io/v2/',
          nativeAsset: {
            name: 'Ethereum',
            symbol: 'ETH',
            decimals: 18,
          },
          blockExplorer: 'https://etherscan.io',
          api: 'https://api.etherscan.io',
        },
      ];
    },
  };
});

jest.mock('provider/providerHooks', () => {
  return {
    useRpcUrls: () => jest.fn(),
  };
});

describe('NetworkModal', () => {
  it('Should match snapshot', () => {
    console.error = jest.fn();
    const { container } = render(
      <NetworkModal isOpen={true} onClose={() => jest.fn()} />,
      {
        container: document.body,
      }
    );
    expect(container).toMatchSnapshot();
  });
  it('Should display chains', () => {
    const { getByText } = render(
      <NetworkModal isOpen={true} onClose={() => jest.fn()} />,
      {
        container: document.body,
      }
    );
    expect(getByText('Ethereum Mainnet')).toBeInTheDocument();
  });
});
