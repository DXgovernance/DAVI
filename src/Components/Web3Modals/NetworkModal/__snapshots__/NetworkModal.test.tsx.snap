// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkModal Should match snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-pGacB doHKro backdropModal"
    >
      <div
        class="sc-fubCzh jCETGP"
      >
        <div
          class="sc-jrAFXE jZmJyP"
        >
          
           
          <div
            class="sc-kEjbQP UrNhf"
          >
            <h2
              class="sc-hKgJUU sc-crrszt iVaMyO gpIzKb"
            >
              <div>
                switchNetwork
              </div>
            </h2>
            <svg
              class="sc-dQoVA faTCxr"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="18"
                x2="6"
                y1="6"
                y2="18"
              />
              <line
                x1="6"
                x2="18"
                y1="6"
                y2="18"
              />
            </svg>
          </div>
          <div
            class="sc-kstqJO glfmLR"
          >
            <div
              class="sc-bBXrwG kFKYkX"
            >
              <div
                class="sc-cxFLGX dwWZCR"
              >
                <div
                  class="sc-iwyWTf gbnIyu"
                >
                  <div
                    class="sc-lmoMya jNRJIs"
                  >
                    <button
                      class="sc-bdfBQB sc-dmlqKv fuUPpr eLvDHF"
                      data-testid=""
                    >
                      <div
                        class="sc-kfzBvY VQpPp"
                      >
                        <span
                          class="sc-hHfuMS jokvNa"
                        />
                        Random Chain
                      </div>
                      <div
                        class="sc-fKFxtB gJFluz"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
   
</body>
`;
